src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs

```diff
@@ -68,20 +68,40 @@ public byte[] ComputeHash(Stream inputStream)
             // Default the buffer size to 4K.
             byte[] buffer = new byte[4096];
             int bytesRead;
+            int bytesRead_firstBlock = 0;
             while ((bytesRead = inputStream.Read(buffer, 0, buffer.Length)) > 0)
             {
+                if (bytesRead_firstBlock == 0)
+                {
+                    bytesRead_firstBlock = bytesRead;
+                }
                 HashCore(buffer, 0, bytesRead);
             }
-            return CaptureHashCodeAndReinitialize();
+            return CaptureHashCodeAndReinitialize(bytesRead_firstBlock);
         }
 
-        private byte[] CaptureHashCodeAndReinitialize()
+        private byte[] CaptureHashCodeAndReinitialize(int inputCount)
```

Comment:
This is for netfx compat to throw when appropriate. In netfx the only way to "clear" after a TransformFinalBlock so it can hash again is to call ComputeHash(empty) which calls Initialize(). It's messy but due to the differences in Initialize + HashFinal between corefx and netfx -- corefx doesn't do anything in Initialize since it does everything in HashFinal thus it wouldn't normally throw when netfx does. Hope that helps, otherwise we can take it offline